plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -841,7 +841,15 @@ def _write_pulp_distribution_file(self, distro_files, old_xml_file_path):
             element.text = distro_file
 
         tree = cElementTree.ElementTree(new_xml_root)
-        tree.write(os.path.join(self.get_working_dir(), constants.DISTRIBUTION_XML))
+        distribution_xml_path = os.path.join(self.get_working_dir(), constants.DISTRIBUTION_XML)
+        try:
+            os.remove(distribution_xml_path)
+        except OSError as e:
+            if e.errno == os.errno.ENOENT:
```

Comment:
I would usually save a couple of lines and do the following, but it's no big deal.

```
if e.errno != os.errno.ENOENT:
    raise
```